lightningd/plugin.c

```diff
@@ -267,13 +273,14 @@ struct plugin *plugin_register(struct plugins *plugins, const char* path TAKES,
 			}
 			if (taken(path))
 				tal_free(path);
+			tal_free(abspath);
 			return NULL;
 		}
 	}
 
 	p = tal(plugins, struct plugin);
 	p->plugins = plugins;
-	p->cmd = tal_strdup(p, path);
+	p->cmd = tal_steal(p, abspath);
```

Comment:
Looks like the `path` is no longer freed (`tal_strdup` would have stolen it instead of copying when `take(path)` was used). This can leak now if it is taken.

Reply:
Isn't `path` already taken earlier in
```
abspath = path_canon(plugins, path);
```

Or maybe not when it failed?
https://github.com/ElementsProject/lightning/blob/9152b8c4240957f185ea026c135eccf335b62069/ccan/ccan/tal/path/path.h#L26-L34

Reply:
Yes, it will take it (below the commented-out code):
https://github.com/ElementsProject/lightning/blob/bad943da55b367158f5e080baea90754aa50c9a7/ccan/ccan/tal/path/path.c#L215-L216